ui/src/components/Pin.tsx

```diff
@@ -33,27 +33,59 @@ export enum PinState {
 }
 
 export interface PinProps {
+  id: string;
+  position: PointTuple;
   state?: PinState;
-  position: [number, number];
+  onPopupOpen?: (id: string) => (event: PopupEvent) => void;
+  onPopupClose?: (event: PopupEvent) => void;
+}
+
+// This is extremely jank
+// TODO: Figure out how anchors actually work
+function getAnchor(state: PinState): PointTuple {
+  switch (state) {
+    case PinState.Resting:
+      return [25, 61];
+    case PinState.Selected:
+      return [29, 70];
+    case PinState.Unfocused:
+      return [22, 57];
```

Comment:
Why is there a different iconAnchor for _unfocused_ compared to _resting_? If I understand it correctly, iconAnchor is where we would like to place the icon image. It defaults to the [lat,lng] we provide being the upper left corner of the svg and then we'd position it accordingly based on image size.